### ISSUE TITLE
DB/Quest: Hunting For Ectoplasm

### ISSUE THREAD

**Issue by TQPS** [TrinityCore/TrinityCore#26737]
### Description

This [quest](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?quest=8924) requires that you activate the [Ectoplasmic Distiller](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?item=21946) device (powered by Goblin Rocket Fuel) in order to obtain items from certain mobs.

The issue is that mobs are dropping the quest item without activating the Distiller.

### Expected behaviour

The device is required to placed down and player is to stand inside the radius and kill required mob. Mob should only drop item if inside the distiller's effect radius.

### Steps to reproduce the problem

1. Accept quest "[Hunting For Ectoplasm](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?quest=8924)"
2. Travel to any of the 3 locations (Silithus, Winterspring or Eastern Plaguelands)
3. Kill required mob in location without using the Distiller
4. They drop the item 

<details>

Example Winterspring

![image](https://user-images.githubusercontent.com/66294837/126749503-73e426d8-b6f6-4296-9b1d-53919d32e19b.png)

</details>

### Branch

3.3.5

### TC rev. hash/commit

TrinityCore rev. 87a4bc38a999 2021-07-23 00:12:58 +0300 (3.3.5 branch) (Win64, RelWithDebInfo, Static) 

### Operating system

Windows 7 x64

### Custom changes

Trickerer's Bots

**Comment by offl** [comment]
I'm afraid it will not work because aura is not even casted by trap

**Comment by Rushor** [comment]
the gameobject could cast the spell every few seconds in sai

**Comment by offl** [comment]
TLDR scripting traps is fine, I was thinking something is broken or missing
```txt
<Offle> spell is set for trap but data5: cooldown (time in seconds) is 0
<Offle> hmm, so traps with no data5: cooldown (time in seconds) are like spell scripts with misvalue=spellid\sound\text\whatever you use in script
<Offle> I was thinking if no cooldown provided, they just repeats spell cast and cooldown is the duration of spell. but cooldown in sniffs is different. so they're just scripted
<%Shauren> Offle: scripted or hidden from client, they do this with npc spells too - no cd on spell data but if you cast it (for example with mind control) then it has a cooldown
<Offle> data5: cooldown hidden from client?
<%Shauren> not yet shipped
<%Shauren> Offle no, i said that they might just set 0 in template sent to client but serverside there is a proper cd
<%Shauren> or its just scripted
<%Shauren> both are possible
<Offle> they also need initial timer
<Offle> in sniff first cast after 2s
```
